structuredheadings/plugin.js

```diff
@@ -594,6 +658,56 @@
             setStyle(editor, node, style);
           }
         }
+      },
+
+      applyPresetToList: {
+        getPresetStyleArray: function (editor, presetName) {
+          if (presetName === "clear") {
+            return [editor.config.listClassMappings.clear];
+          }
+
+          var styleArray = presetName.split(".").map(function (str) {
+            var key = str.trim()[0];
+            return editor.config.listClassMappings[key];
+          }).filter(function (str) {
+            if (str) {
+              // also purge empty strings, undef, etc. because I am lazy
+              return true;
+            } else {
+              return false;
+            }
+          });
+
+          return styleArray;
+        },
+        numListsInPath: function (elementPath) {
```

Comment:
an elementpath consists of an element and all the parents of it up to the root node (either body or something else you specify) 

so this really just counts the ols